### PR TITLE
Fix: Raise an exception when trying to create a smart collection/playlist with items

### PR DESCRIPTION
**Pull request by glensc**
## Description

Fixes https://github.com/pkkid/python-plexapi/issues/1375

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the docstring for new or existing methods
- [ ] I have added tests when applicable

### CODE REVIEW

**Comment by glensc** [comment]
actually, i'm not sure what was happening. as the "items" doesn't seem to be passed to the `_createSmart` method

**Comment by glensc** [comment]
so, what was happening was that it called this URL:

- `/playlists?smart=1&title=The Melchior Collection&type=video&uri=server://[redacted]/com.plexapp.plugins.library/library/sections/1/all?includeGuids=1&type=1`


which seems like smart playlist with no filters (so it contains whole library)

**Comment by glensc** [comment]
Maybe what could be done here instead is to raise an error if no filters are provided

**Comment by glensc** [comment]
So the change here should be fine. It would disallow using smart and items, possibly not confusing the user. 

user can still create smart playlist without filters as long as they don't also pass items.

File: plexapi/playlist.py [hunk 1]
```diff
@@ -434,6 +434,8 @@ def create(cls, server, title, section=None, items=None, smart=False, limit=None
         if m3ufilepath:
             return cls._createFromM3U(server, title, section, m3ufilepath)
         elif smart:
+            if items:
+                raise BadRequest('Cannot fill items for a smart playlist.')
```
Comment (JonnyWong16) [inline plexapi/playlist.py:438]:
```suggestion
                raise BadRequest('Cannot create a smart playlist with items.')
```

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]
Could you also add the same exception for smart collections?

https://github.com/pkkid/python-plexapi/blob/abcab4f78d22e86b380327c6014d3148e0248e6d/plexapi/collection.py#L505-L506

**Review by JonnyWong16 — APPROVED** [review]

**Pull request merged**